platform/viewer/src/connectedComponents/ToolbarRow.js

```diff
@@ -128,7 +134,15 @@ class ToolbarRow extends Component {
           </div>
           {buttonComponents}
           <ConnectedLayoutButton />
-          <ConnectedPluginSwitch />
+          {this.state.isPlugSwitchOn ?
+            <ConnectedExitPluginSwitch
+              togglePlugSwitchStatus={this.togglePlugSwitchStatus}
+            />
+            :
+            <ConnectedPluginSwitch
+              togglePlugSwitchStatus={this.togglePlugSwitchStatus}
+            />
```

Comment:
Instead of toggling the component, can we toggle a prop on the component that changes the icon / label text / action?

Reply:
I agree. I had thought of keeping the toggleStatus private to the ConnectedPluginSwitch component and let it change the onClick action and icon/label, so as to become a toggling button in itself. But, that would directly alter the function of ConnectedPluginSwitch and I wasn't sure if that was acceptable.